include/deal.II/base/thread_local_storage.h

```diff
@@ -19,22 +19,56 @@
 
 #  include <deal.II/base/config.h>
 
-DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
-#  ifdef DEAL_II_WITH_THREADS
-#    include <tbb/enumerable_thread_specific.h>
-#  endif
-DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
-
+#  include <deal.II/base/exceptions.h>
 
+#  include <list>
+#  include <map>
+#  include <shared_mutex>
+#  include <thread>
+#  include <vector>
```

Comment:
list, map, AND vector?

Reply:
Jup. We need the map because we use it and we need to import vector and list for the workaround found below (it is not possible to forward declare these data types in a portable way). On the bright side - we already include list and vector in all of our compilation units, except for the ThreadLocalStorage tests.

Those also only fail for gcc-5 only, all other compiler seem to already include vector and list.